envy.sh

```diff
@@ -20,37 +79,45 @@ envy() {
         CONTENTS=$(cat "${FILENAME}" | grep "^[^#]")
     fi
     while read -r PAIR; do
-        K=$(echo ${PAIR} | sed 's/\([^=]*\)=\(.*\)/\1/')
-        V=$(echo ${PAIR} | sed 's/\([^=]*\)=\(.*\)/\2/')
+        K=$(sed 's/\([^=]*\)=\(.*\)/\1/' <<< "${PAIR}")
+        V=$(sed 's/\([^=]*\)=\(.*\)/\2/' <<< "${PAIR}")
+        # Check if templating
+        if grep -q "{{.*}}" <<< "${V}"; then
```

Comment:
Seeing quite a few references to .* representing the variable token between the {{ and }} delimiters. Does this mean that someone could potentially have {{}} in the template?
Maybe references could be more specific to be something like `[a-zA-Z0-9-_]\+` or whatever the valid definition of a variable is :)

Reply:
This is just a basic check to see if there is any templating logic to know if it should even bother to run the templating

Reply:
Understood, although I meant for my comment to be more generic referring to the '.*' across the board for things like pattern matching or substitution e.g. `sed 's/\([^=]*\)=\(.*\)/\1/')`. Is there any reason why we shouldn't consider replacing these with something more concrete that enforces 1..* characters (e.g. `.\+`) rather than 0..* (`.*`)?

Reply:
Using Bash parsing like in mush
```
K=${PAIR%%"="*}
V=${PAIR#*"="*}
```

Reply:
👍